ldms/src/core/ldms.c

```diff
@@ -338,9 +340,11 @@ static const char *perm_string(uint32_t perm)
 	return str;
 }
 
-char *set_state(struct ldms_set *set)
+#define STATE_BUF_SZ 8
+typedef char(*state_buf)[STATE_BUF_SZ];
+static const char *set_state(struct ldms_set *set, state_buf sstr)
```

Comment:
Same here pass in the char *, and the return type should not be const. The caller is passing in the buffer, and the function is modifying it.